src/org/traccar/protocol/WondexProtocolDecoder.java

```diff
@@ -110,7 +110,7 @@ protected Object decode(
             position.set(Position.KEY_SATELLITES, satellites);
 
             position.set(Position.KEY_EVENT, parser.next());
-            position.set(Position.KEY_BATTERY, parser.next());
+            position.set(Position.KEY_BATTERY, parser.nextDouble(0));
```

Comment:
For example here is seems optional, so it's better to use `parser.nextDouble()`. Please check all the places.